sdk/database/helper/connutil/sql.go

```diff
@@ -59,30 +59,19 @@ func (c *SQLConnectionProducer) Init(ctx context.Context, conf map[string]interf
 		return nil, fmt.Errorf("connection_url cannot be empty")
 	}
 
-	// Do not allow the username or password template pattern to be used as
-	// part of the user-supplied username or password
-	if strings.Contains(c.Username, "{{username}}") ||
-		strings.Contains(c.Username, "{{password}}") ||
-		strings.Contains(c.Password, "{{username}}") ||
-		strings.Contains(c.Password, "{{password}}") {
-
-		return nil, fmt.Errorf("username and/or password cannot contain the template variables")
-	}
-
-	// Don't escape special characters for MySQL password
-	// Also don't escape special characters for the username and password if
-	// the disable_escaping parameter is set to true
+	// Don't escape special characters for the username and password if the
+	// disable_escaping parameter is set to true. A user does this if the
+	// connection_url is actually not a URL, but database-specific other
+	// format, typically space or semicolon separated key=value pairs.
 	username := c.Username
 	password := c.Password
 	if !c.DisableEscaping {
 		username = url.PathEscape(c.Username)
-	}
-	if (c.Type != "mysql") && !c.DisableEscaping {
```

Comment:
This mysql special case can be deleted - the mysql plugin is no longer using this code at all. As can be seen in the first file touched in this PR, it now has its own forked copy.